QuickConnectPlugin/QuickConnectPluginExt.cs

```diff
@@ -193,6 +194,44 @@ public class QuickConnectPluginExt : Plugin {
                 );
                 this.menuItems.Add(menuItem);
             };
+            if (hostPwEntry.ConnectionMethods.Contains(ConnectionMethodType.WinSCP))
+            {
+                var winScpPath = !String.IsNullOrEmpty(this.Settings.WinScpPath) ? this.Settings.WinScpPath : QuickConnectUtils.GetWinScpPath();
+                var winScpConsoleMenuItem = new ToolStripMenuItem()
+                {
+                    Text = OpenWinScpMenuItemText,
+                    Image = (System.Drawing.Image)QuickConnectPlugin.Properties.Resources.winscp,
+                    Enabled = hostPwEntry.HasIPAddress && !String.IsNullOrEmpty(winScpPath)
+                };
+                winScpConsoleMenuItem.Click += new EventHandler(
+                    delegate (object obj, EventArgs ev) {
+                        try
+                        {
+                            // Start a detached process.
+                            Process cmd = new Process();
+                            cmd.StartInfo.FileName = "cmd.exe";
+                            cmd.StartInfo.RedirectStandardInput = true;
+                            cmd.StartInfo.RedirectStandardOutput = true;
+                            cmd.StartInfo.CreateNoWindow = true;
+                            cmd.StartInfo.UseShellExecute = false;
+                            cmd.Start();
+                            cmd.StandardInput.WriteLine(String.Format("\"{0}\" scp://{2}:{3}@{1}",
+                                winScpPath,
+                                hostPwEntry.IPAddress,
+                                hostPwEntry.GetUsername(),
+                                hostPwEntry.GetPassword())
+                                );
+                            cmd.StandardInput.Flush();
+                            cmd.StandardInput.Close();
```

Comment:
There are a few thing that need to be done here:
- create a new class `WinSCPArgumentsFormatter` implementing `IArgumentsFormatter`
- add a test case for `WinSCPArgumentsFormatter`
- refactor using `ProcessUtils.StartDetached()`